services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudV2Spec.groovy

```diff
@@ -1066,6 +1068,123 @@ class FlowCrudV2Spec extends HealthCheckSpecification {
+        with(northbound.pingFlow(flow.flowId, new PingInput())) {
+            it.forward.pingSuccess
+            it.reverse.pingSuccess
+        }
+
+        and: "The src switch passes switch validation"
+        with(northbound.validateSwitch(srcSwitch.dpId)) { validation ->
+            validation.verifyRuleSectionsAreEmpty(["missing", "excess", "misconfigured"])
+            validation.verifyMeterSectionsAreEmpty(["missing", "excess", "misconfigured"])
+        }
+        def srcSwitchIsFine = true
+
+        when: "Update the flow: switch id on the dst endpoint"
+        def newDstSwitch = allSwitches[2]
+        flowHelperV2.updateFlow(flow.flowId, flow.tap {
+            it.destination.switchId = newDstSwitch.dpId
+            it.destination.portNumber = newPortNumber
+        })
+
+        then: "Flow is really updated"
+        with(northbound.getFlow(flow.flowId)) {
+            it.destination.switchDpId == newDstSwitch.dpId
+        }
+
+        and: "Flow rules are removed from the old dst switch"
+        def flowInfoFromDb3 = database.getFlow(flow.flowId)
+        Wrappers.wait(RULES_DELETION_TIME) {
+            with(northbound.getSwitchRules(dstSwitch.dpId).flowEntries.findAll {
+                !Cookie.isDefaultRule(it.cookie)
+            }) { rules ->
+                rules.findAll {
+                    it.cookie in [flowInfoFromDb2.forwardPath.cookie.value, flowInfoFromDb2.reversePath.cookie.value]
+                }.empty
+            }
+        }
+
+        and: "Flow rules are installed on the new dst switch"
+        Wrappers.wait(RULES_INSTALLATION_TIME) {
+            with(northbound.getSwitchRules(newDstSwitch.dpId).flowEntries.findAll {
+                !Cookie.isDefaultRule(it.cookie)
+            }) { rules ->
+                rules.findAll {
+                    it.cookie in [flowInfoFromDb3.forwardPath.cookie.value, flowInfoFromDb3.reversePath.cookie.value]
+                }.size() == 2
+            }
+        }
+
+        and: "Flow is valid and pingable"
+        northbound.validateFlow(flow.flowId).each { direction -> assert direction.asExpected }
+        with(northbound.pingFlow(flow.flowId, new PingInput())) {
+            it.forward.pingSuccess
+            it.reverse.pingSuccess
+        }
+
+        and: "The new and old dst switches pass switch validation"
+        [dstSwitch, newDstSwitch]*.dpId.each { switchId ->
+            with(northbound.validateSwitch(switchId)) { validation ->
+                validation.verifyRuleSectionsAreEmpty(["missing", "excess", "misconfigured"])
+                validation.verifyMeterSectionsAreEmpty(["missing", "excess", "misconfigured"])
+            }
+        }
+        def dstSwitchesAreFine = true
+
+        cleanup:
+        flowHelperV2.deleteFlow(flow.flowId)
+        !srcSwitchIsFine && northbound.synchronizeSwitch(srcSwitch.dpId, true)
+        !dstSwitchesAreFine && [dstSwitch, newDstSwitch]*.dpId.each { northbound.synchronizeSwitch(it, true) }
```

Comment:
Not critical, but you may also want a null check for `dstSwitch` and `newDstSwitch` vars, since they may not yet be available. Can be omitted though, for the sake of having less code

Reply:
the check was added
I guess we should be confident in cleanup)
otherwise the is no sense to mark this test as `Tidy`